wordpress-varnish-as-a-service.php

```diff
@@ -59,6 +59,10 @@ public function __construct() {
 		add_action('wp_delete_nav_menu', array(&$this, 'WPVarnishPurgeAll'), 99);
 		add_action('wp_add_nav_menu_item', array(&$this, 'WPVarnishPurgeAll'), 99);
 		add_action('wp_update_nav_menu_item', array(&$this, 'WPVarnishPurgeAll'), 99);
+		add_action('update_option_category_base', array(&$this, 'WPVarnishPurgeAll'), 99);
+		add_action('update_option_tag_base', array(&$this, 'WPVarnishPurgeAll'), 99);
+		add_action('permalink_structure_changed', array(&$this, 'WPVarnishPurgeAll'), 99);
```

Comment:
I think that hooking into both the `permalink_structure_changed` and `update_option_permalink_structure` actions is going to result in bans happening twice when permalink structures are changed:  https://core.trac.wordpress.org/browser/tags/5.4/src/wp-includes/class-wp-rewrite.php#L1881

Only one of these two action hooks should be necessary.

Reply:
You're right. I wasn't sure if that was necessary researching different options.

Found out that WP Rocket plugin was open source on Github, and this is their cache clearing:
https://github.com/wp-media/wp-rocket/pull/1437/files